### ISSUE TITLE
Need a way to temporarily release unused GPU memory without destroying nodes

### ISSUE THREAD

**Issue by brianchirls** [brianchirls/Seriously.js#81]
Sometimes large compositions with multiple scenes can cause problems running on devices without great GPU resources, like a mobile phone/tablet. For example, when running [this piece](http://campcentury.github.io/campcentury/) on a MacBook Air, WebGL crashes (lost context) when switching between scenes.

I propose to add `.purge()` and `.restore()` methods to all node types and to the `Seriously` composition object for releasing all GPU resources (shaders, textures, frame buffers, etc.). All the resources will be rebuilt when `.restore()` is called or automatically when any purged nodes need to be rendered.

**Comment by darkyen** [comment]
I am worried about the performance implecation of purge and restore methods, are they supposed to be called in realtime ?

**Comment by brianchirls** [comment]
No, they're supposed to be called outside of "hot" render cycles, when
you're shutting down or spinning up sections of a composition.

**Comment by darkyen** [comment]
Ah then they definitely make sense.
 :D
